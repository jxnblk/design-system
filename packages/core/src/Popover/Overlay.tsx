import React from 'react'
import styled from 'styled-components'
import { Box } from '../Box/Box'

const StyledOverlay = styled(Box)`
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  opacity: ${({ overlayOpacity }) => overlayOpacity};
  cursor: pointer;
  display: ${({ popoverOpen }) => (popoverOpen ? 'block' : 'none')};
`

export type OverlayProps = {
  overlayOpacity?: number
  zIndex?: number
}

const Overlay = ({ overlayOpacity, isOpen }) => {
  return (
    <StyledOverlay
      aria-hidden='true'
      tabIndex='-1'
      popoverOpen={isOpen}
      overlayOpacity={overlayOpacity}
      color='background.darkest'
    />
  )
}

export default Overlay
