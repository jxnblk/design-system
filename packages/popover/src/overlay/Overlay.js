import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import { Link } from 'pcln-design-system'

const StyledOverlay = styled(Link)`
  display: block;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  opacity: ${({ popoverOpen, overlayOpacity }) =>
    popoverOpen ? overlayOpacity : 0};
  z-index: ${({ zIndex }) => (zIndex < 0 ? 0 : zIndex)};
  cursor: pointer;
`

const Overlay = ({
  className,
  popoverOpen,
  handleClick,
  bg,
  overlayOpacity,
  zIndex
}) => {
  return (
    <StyledOverlay
      className={className}
      onClick={handleClick}
      popoverOpen={popoverOpen}
      bg={bg}
      overlayOpacity={overlayOpacity}
      zIndex={zIndex}
    />
  )
}

Overlay.propTypes = {
  className: PropTypes.string,
  popoverOpen: PropTypes.bool,
  handleClick: PropTypes.func,
  bg: PropTypes.string,
  overlayOpacity: PropTypes.number,
  zIndex: PropTypes.number
}

Overlay.defaultProps = {
  popoverOpen: true,
  bg: 'black',
  overlayOpacity: 0,
  zIndex: 101
}

export default Overlay
