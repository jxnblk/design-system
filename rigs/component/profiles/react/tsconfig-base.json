{
  "$schema": "http://json.schemastore.org/tsconfig",

  "compilerOptions": {
    "outDir": "../../../../../lib-esm",
    "rootDir": "../../../../../src",
    "rootDirs": ["../../../../../src", "../../../../../temp/sass-ts"],

    "forceConsistentCasingInFileNames": true,
    "jsx": "react",
    "declaration": true,
    "sourceMap": true,
    "declarationMap": true,
    "inlineSources": true,
    "experimentalDecorators": true,

    "esModuleInterop": true,
    "noEmitOnError": false,
    "allowUnreachableCode": false,

    "module": "es2015",
    "moduleResolution": "node",
    "target": "es5",
    "lib": ["es5", "scripthost", "es2015.collection", "es2015.promise", "es2015.iterable", "dom", "es2016"],

    // deviations from @rushstack/heft-web-rig/profiles/library/tsconfig-base.json
    "strict": false,

    "types": [
      "../../rigs/component/node_modules/@types/heft-jest",
      "../../rigs/component/node_modules/@types/node"
    ],

    "allowJs": true
  },
  "include": ["../../../../../src/**/*"],
  "exclude": ["node_modules", "lib-esm"]
}
