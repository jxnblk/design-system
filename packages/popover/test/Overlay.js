import React from 'react'
import { fireEvent } from '@testing-library/react'
import Overlay from '../src/Overlay'

const overlayProps = {
  overlayOpacity: 0.3,
  zIndex: 200,
}

describe('Background Overlay', () => {
  test('Active overlay', () => {
    const { container } = renderWithTheme(
      <Overlay {...overlayProps} popoverOpen={true} />
    )
    expect(container).toMatchSnapshot()
  })

  test('Inactive overlay', () => {
    const { container } = renderWithTheme(
      <Overlay {...overlayProps} popoverOpen={false} />
    )
    expect(container).toMatchSnapshot()
  })

  test('Handle click event', () => {
    const onClick = jest.fn()
    const { container } = renderWithTheme(
      <Overlay {...overlayProps} popoverOpen={true} handleClick={onClick} />
    )
    fireEvent.click(container.firstChild)
    expect(onClick).toHaveBeenCalled()
  })
})
