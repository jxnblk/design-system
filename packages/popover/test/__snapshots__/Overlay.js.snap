// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background Overlay Active overlay 1`] = `
.c0 {
  display: block;
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background-color: #000;
  opacity: 0.3;
  z-index: 200;
  cursor: pointer;
}

<div>
  <div
    aria-hidden="true"
    class="c0 "
    tabindex="-1"
  />
</div>
`;

exports[`Background Overlay Inactive overlay 1`] = `
.c0 {
  display: block;
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background-color: #000;
  opacity: 0;
  z-index: 200;
  cursor: pointer;
}

<div>
  <div
    aria-hidden="true"
    class="c0 "
    tabindex="-1"
  />
</div>
`;
